dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphHostControl.xaml.cs

```diff
@@ -0,0 +1,197 @@
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+// 
+//  The MIT License (MIT)
+// 
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+// 
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+// 
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//  ---------------------------------------------------------------------------------
+
+using BarGraphUtility;
+using SharpDX;
+using SharpDX.Direct2D1;
+using SharpDX.DXGI;
+using SharpDX.Mathematics.Interop;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+using SysWin = System.Windows;
+using Windows.UI.Composition;
+
+
+namespace VisualLayerIntegration
+{
+    /// <summary>
+    /// Interaction logic for BarGraphHostControl.xaml
+    /// </summary>
+    public partial class BarGraphHostControl : UserControl
+    {
+        private CompositionHost compositionHost;
+        private Compositor compositor;
+        private Windows.UI.Composition.ContainerVisual graphContainer;
+        private BarGraph currentGraph;
+
+        private double currentDpiX = 96.0;
+        private double currentDpiY = 96.0;
+
+        protected WindowRenderTarget windowRenderTarget;
+        private static RawColor4 white = new RawColor4(255, 255, 255, 255);
+
+        public BarGraphHostControl()
+        {
+            InitializeComponent();
+            Loaded += BarGraphHostControl_Loaded;
+            DataContextChanged += BarGraphHostControl_DataContextChanged;
+        }
+
+        private void BarGraphHostControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            UpdateGraph();
+        }
+
+        private void BarGraphHostControl_Loaded(object sender, RoutedEventArgs e)
+        {
+            // If the user changes the DPI scale setting for the screen the app is on,
+            // the CompositionHostControl is reloaded. Don't redo this set up if it's
+            // already been done.
+            if (compositionHost is null)
```

Comment:
What does "is null" mean here? Why not (compositionHost == null) ?

Reply:
updated